src/App.tsx

```diff
@@ -1,17 +1,123 @@
-import React from 'react';
-import './App.css';
+import React, { useState } from 'react';
+import './App.scss';
 
+import todos from './api/todos';
 import users from './api/users';
+import { TodoList } from './components/TodoList';
+import { PreparedTodo } from './typedefs';
+
+const preparedTodos: PreparedTodo[] = todos.map(todo => {
+  return {
+    ...todo,
+    user: users.find(user => user.id === todo.userId) || null,
+  };
+});
 
 const App: React.FC = () => {
+  const [todosList, setTodosList] = useState([...preparedTodos]);
+  const [title, setTitle] = useState('');
+  const [selectUser, setSelectUser] = useState(0);
+  const [completed, setCompleted] = useState(false);
+  const [errorUser, setErrorUser] = useState(false);
+  const [errorTitle, setErrorTitle] = useState(false);
+
+  const addTodo = (event: React.FormEvent) => {
+    event.preventDefault();
+    if (title && selectUser) {
+      setTodosList([
+        ...todosList,
+        {
+          userId: selectUser,
+          id: todosList.length + 1,
+          title,
+          completed,
+          user: users.find(user => user.id === selectUser),
+        },
+      ]);
+      setTitle('');
+      setCompleted(false);
+      setSelectUser(0);
+    }
+
+    setErrorUser(!selectUser);
+    setErrorTitle(!title);
+  };
+
   return (
     <div className="App">
-      <h1>Add todo form</h1>
+      <h1 className="title">Add todo form</h1>
+      <form
+        action="get"
+        onSubmit={addTodo}
+        className="form"
+      >
+        <div>
+          <select
+            className="form__select"
+            name=""
+            id=""
+            data-cy="userSelect"
+            placeholder="Choose a user"
+            value={selectUser}
+            onChange={(changeEvent) => {
+              setSelectUser(+changeEvent.target.value);
+              setErrorUser(false);
+            }}
+          >
+            <option
+              value={0}
+              disabled
```

Comment:
you could also add `hidden` so this option is not even visible on the list
```suggestion
              disabled
              hidden
```